staging/src/k8s.io/client-go/tools/cache/fifo.go

```diff
@@ -294,11 +294,8 @@ func (f *FIFO) Pop(process PopProcessFunc) (interface{}, error) {
 		if f.initialPopulationCount > 0 {
 			f.initialPopulationCount--
 		}
-		item, ok := f.items[id]
-		if !ok {
-			// Item may have been deleted subsequently.
-			continue
-		}
+
+		item := f.items[id]
```

Comment:
As mentioned, the comment is inccorect. See:

https://github.com/kubernetes/kubernetes/blob/b1b93e30131986319f1d46a62771feb9c0f097ae/staging/src/k8s.io/client-go/tools/cache/fifo.go#L210-L220